hbase-shell/src/main/ruby/hbase/table.rb

```diff
@@ -243,7 +243,7 @@ def _deleteall_internal(row, column = nil,
                             args = {}, all_version = true)
       # delete operation doesn't need read permission. Retaining the read check for
       # meta table as a part of HBASE-5837.
-      if is_meta_table?
+      if is_meta_table? and !(row.is_a?(Hash) and row.key?('ROWPREFIXFILTER'))
```

Comment:
It's dangerous to delete rows in meta table by 'ROWPREFIXFILTER', right? If the rowPrefix matches some rows unexpected, the meta data will be deleted unexpected. I think maybe this is why the origin design seperated meta table here.

Reply:
Yes, I have this concen too. Let me fix the error message?

Reply:
Of course, you can try to make the descriptions more clearly.